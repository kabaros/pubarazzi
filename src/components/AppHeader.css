
.app-header {
    background-color: #333333;
    padding: 10px;
    color: white;
    position: relative;
}

.close-button {
    right: 10px;
    top: 10px;
    font-size: 18px;
    font-weight: bold;
    text-align: right;
    position: absolute
}