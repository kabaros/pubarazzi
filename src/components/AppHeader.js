import React from 'react';
import './AppHeader.css';

export default ({ title, closeButton = () => {} }) => {
    if (title) {
        return (<header className="app-header">
            {title}
            <div className="close-button" onClick={closeButton}>x</div>
        </header>)
    }
    return <header className="app-header">Where do you want to go today?</header>
}