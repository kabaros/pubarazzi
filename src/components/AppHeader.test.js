import React from 'react';
import { render } from 'react-dom';
import AppHeader from './AppHeader';

it('renders default title', () => {
    const div = document.createElement('div');
    render(<AppHeader />, div);

    expect(div).toMatchSnapshot();
});


it('renders passed title', () => {
    const div = document.createElement('div');
    render(<AppHeader title="Something" />, div);

    expect(div).toMatchSnapshot();
});
