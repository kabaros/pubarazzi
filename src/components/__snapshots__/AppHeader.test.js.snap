// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders default title 1`] = `
<div>
  <header
    class="app-header"
  >
    Where do you want to go today?
  </header>
</div>
`;

exports[`renders passed title 1`] = `
<div>
  <header
    class="app-header"
  >
    Something
    <div
      class="close-button"
    >
      x
    </div>
  </header>
</div>
`;
